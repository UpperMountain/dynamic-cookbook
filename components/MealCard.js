import React from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { BlurView } from "expo";
var s = require("./Styles");

const ImageContainer = props => {
  let pad = props.width == 210 ? 4 : 0;
  return (
    <View
      style={{
        width: props.width,
        height: props.height,
        overflow: "hidden",
        backgroundColor: "white",
        borderRadius: 19
      }}
    >
      {props.children}
      <View
        style={{
          position: "absolute",
          zIndex: 1,
          top: props.height * 0.78 + pad,
          width: props.width,
          height: props.height * 0.22 - pad,
          backgroundColor: "rgba(255, 255, 255, .6)"
        }}
      >
        <BlurView
          tint={"light"}
          intensity={65}
          style={{
            width: props.width,
            height: props.height * 0.22 - pad,
            paddingLeft: 16,
            paddingTop: 4 + pad
          }}
        >
          <Text style={[s.body, { marginBottom: pad }]}>Pancakes</Text>
          <Text style={s.caption}>20min | serves 4</Text>
        </BlurView>
      </View>
    </View>
  );
};

export class MealCard extends React.Component {
  constructor(props) {
    super();
    this.state = {};
  }

  render() {
    return (
<<<<<<< HEAD
      <View style={style.shadow}>
=======
      <View style={s.shadow}>
>>>>>>> general progress
        <ImageContainer width={210} height={262}>
          <Image
            style={{ width: 210, height: 262 }}
            source={require("../assets/images/pancakes.jpg")}
          />
        </ImageContainer>
      </View>
    );
  }
}

export class SmallMealCard extends React.Component {
  constructor(props) {
    super();
    this.state = {};
  }

  render() {
    return (
<<<<<<< HEAD
      <View style={style.shadow}>
=======
      <View style={s.shadow}>
>>>>>>> general progress
        <ImageContainer width={142} height={191}>
          <Image
            style={{ width: 142, height: 191 }}
            source={require("../assets/images/pancakes.jpg")}
          />
        </ImageContainer>
      </View>
    );
  }
}
