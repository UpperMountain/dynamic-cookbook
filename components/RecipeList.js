import React from "react";
import { ScrollView } from "react-native";

export default (RecipeList = props => {
  const itemWidth = props.small ? 162 : 230;

  return (
    <ScrollView
      horizontal
      decelerationRate={0}
      snapToInterval={itemWidth}
      snapToAlignment={"center"}
    >
      {props.children}
    </ScrollView>
  );
});
