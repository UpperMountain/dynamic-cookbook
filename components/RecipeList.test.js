import React from "react";
import RecipeList from "./RecipeList";
import renderer from "react-test-renderer";
import { MealCard } from "./MealCard";

it("should render something without exploding", () => {
  const cmp = renderer.create(<RecipeList />);
});

it("should render a list with two meal cards", () => {
  const cmp = renderer.create(
    <RecipeList>
      <MealCard />
      <MealCard />
    </RecipeList>
  );
  expect(cmp.toJSON()).toMatchSnapshot();
});
