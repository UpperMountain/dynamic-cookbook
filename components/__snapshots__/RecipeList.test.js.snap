// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list with two meal cards 1`] = `null`;
